[Modules] Support for inGuest Patch support with Azure Update Manager

jdrepo opened the pull request:
# Description

Based on #3954 

</details>

Support for the "installPatches" property in the Microsoft.Maintenance/maintenanceConfigurations resource

### Pipeline references
<!-- For module/pipeline changes, please create and attach the status badge of your successful run. -->

| Pipeline |
| - |
|[![Maintenance - MaintenanceConfigurations](https://github.com/jdrepo/ResourceModules/actions/workflows/ms.maintenance.maintenanceconfigurations.yml/badge.svg)](https://github.com/jdrepo/ResourceModules/actions/workflows/ms.maintenance.maintenanceconfigurations.yml) |

## Type of Change

<!-- Use the check-boxes [x] on the options that are relevant. -->

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Update to documentation

## Checklist

<!-- Use the check-boxes [x] on the options that are relevant. -->

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings
- [x] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (readme)
- [ ] I did format my code

AlexanderSehr reviewed (changes requested):
Looks great. 2 asks though:
- Please also update the readme using our Set-ModuleReadme utility
- If possible, it would also be great if we could add the usage of the parameter to the common test case (which would also need a subsequent update of the Readme). This helps us to make sure the change works and further is a good reference for the community on how the object must look like

jdrepo commented:
Added missing module README.md and modified common test case. Can you have a look please ?

Another common question: How can I assign the follwong labels to PR:

-  [cat] modules 
- enhancement

AlexanderSehr commented:
> Added missing module README.md and modified common test case. Can you have a look please ?
> 
> Another common question: How can I assign the follwong labels to PR:
> 
> * [cat] modules
> * enhancement

You don't, but I just did. Will review shortly :)

AlexanderSehr commented on modules/maintenance/maintenance-configuration/.test/common/main.test.bicep:
```diff
@@ -65,15 +67,32 @@ module testDeployment '../../main.bicep' = {
     }
     namespace: '${serviceShort}ns'
     visibility: 'Custom'
+
```
```suggestion
```

AlexanderSehr commented:
Looks great. Just 1 comment, then we can merge. Thanks @jdrepo

AlexanderSehr commented:
Here we go. Thanks @jdrepo

jdrepo commented:
Thanks. 
What will happen to the referenced issue #3954  in the Product backlog ? 
Will it be closed automatically or should I close it by myself ?
